cli/args/lockfile.rs

```diff
@@ -109,23 +107,9 @@ impl CliLockfile {
 
     let filename = match flags.lock {
       Some(ref lock) => PathBuf::from(lock),
-      None => match maybe_config_file {
-        Some(config_file) => {
-          if config_file.specifier.scheme() == "file" {
-            match config_file.resolve_lockfile_path()? {
-              Some(path) => path,
-              None => return Ok(None),
-            }
-          } else {
-            return Ok(None);
-          }
-        }
-        None => match maybe_package_json {
-          Some(package_json) => {
-            package_json.path.parent().unwrap().join("deno.lock")
-          }
-          None => return Ok(None),
-        },
+      None => match workspace.resolve_lockfile_path()? {
+        Some(path) => path,
+        None => return Ok(None),
```

Comment:
By the way, @nathanwhit, we should update https://github.com/denoland/deno_config/pull/84/files to get the frozen flag at the same time as resolving the lockfile path from the workspace.